### PR TITLE
chore: added room type in create a room and the column in the database

### LINKED ISSUE #320: add a room type to each room
add a room type in the UI and in the database

### What needs to be done
add a new collum in the database for the room type 
add a new variable attached to each room called room type
make the admin able to set the room type when creating a room

### PR DESCRIPTION
**Pull request by joeyabou**
Added a room type variable to rooms. Modified the create room UI to reflect this change and added the collum of room type to the database.

### Test plan (required)

image

### Closing issues
- closes #320 

### Sub-tasks Checklist
- [X] UI Implementation
- [ ] UI Test
- [X] Backend Implementation
- [ ] Backend Tests
- [ ] Integration Tests
- [ ] System Tests
- [ ] Documentation

### CODE REVIEW

File: resources/js/Pages/Admin/Rooms/UpdateRoomForm.vue [hunk 1]
```diff
@@ -69,6 +69,19 @@
                 />
             </div>
 
+             <div class="m-6">
+                <jet-label for="room_type" value="room_type" />
+                <select v-model="form.status" class="mt-1 block w-full" name="room_type" id="room_typeUpdate">
+                    <option :value="form.status" selected="selected">{{form.status}}</option>
+                    <option v-if="form.status != 'available'" :value="'available'">available</option> 
+                    <option v-else :value="'unavailable'">anavailable</option>
```
Comment (WindOfRussia) [inline resources/js/Pages/Admin/Rooms/UpdateRoomForm.vue:77]: this looks like this was a copy paste of status. make sure you add the proper room type options here

**Review by WindOfRussia — CHANGES_REQUESTED** [review]

**Review by Alex-Hayeur — APPROVED** [review]
The changes look good after addressing andrew's change request. The migration also has the down function which is something that's been forgotten a few times so good job there.

Should be good to merge.

**Review by WindOfRussia — APPROVED** [review]

**Review by Alex-Hayeur — APPROVED** [review]
Passing along previous room now that tests are fixed.

File: database/factories/RoomFactory.php [hunk 2]
```diff
@@ -27,6 +27,7 @@ public function definition()
             'floor' => $this->faker->RandomDigit,
             'building' => $this->faker->word,
             'status' => $this->faker->randomElement(["available", "unavailable"]),
+            'room_type' => $this->faker->randomElement(["lounge", "mezzanine","conference"]),
```
Comment (alexstojda) [inline database/factories/RoomFactory.php:30]: Can we have these types defined centrally somewhere? Maybe as a `config` or as a static field on the Model?  @WindOfRussia is there a best practice for this?

File: resources/js/Pages/Admin/Rooms/UpdateRoomForm.vue [hunk 3]
```diff
@@ -71,6 +71,19 @@
                     />
                 </div>
 
+             <div class="m-6">
+                <jet-label for="room_type" value="room_type" />
+                <select v-model="form.room_type" class="mt-1 block w-full" name="room_type" id="room_typeUpdate">
+                    <option value="lounge" :selected="form.room_type=='lounge'">Lounge</option>
```
Comment (alexstojda) [inline resources/js/Pages/Admin/Rooms/UpdateRoomForm.vue:77]: Load these options from wherever they will be stored (see previous comment)

File: app/Http/Controllers/RoomController.php [hunk 4]
```diff
@@ -61,6 +61,7 @@ public function store(Request $request)
             'ambiant_music' => ['required', 'boolean'],
             'sale_for_profit' => ['required', 'boolean'],
             'fundraiser' => ['required', 'boolean'],
+            'room_type' => ['required', 'string', 'max:255'],
```
Comment (alexstojda) [inline app/Http/Controllers/RoomController.php:64]: Plz validate that its one of the allowed options here

**Review by alexstojda — CHANGES_REQUESTED** [review]
Nice work, just some minor things

**Review by alexstojda — CHANGES_REQUESTED** [review]
Also, please validate your commit messages follow our standards.

File: database/factories/RoomFactory.php (hunk 2 above)
Comment (WindOfRussia) [reply database/factories/RoomFactory.php:30]: normally enums are used. but we're in php sooo the best next thing is to add static string variables to the Model.

File: resources/js/Pages/Admin/Rooms/UpdateRoomForm.vue (hunk 3 above)
Comment (WindOfRussia) [reply resources/js/Pages/Admin/Rooms/UpdateRoomForm.vue:77]: loading these from backend will be a little complicated. basically need to update the controller to send a availableRoomTypes prop to frontend

**Comment by alexstojda** [comment]
Commit messages are still malformed: 
image

I need to tweak the commit validation step

**Pull request closed without merging**